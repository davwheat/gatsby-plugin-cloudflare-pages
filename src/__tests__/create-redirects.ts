import { readFile } from 'fs-extra'

import createRedirects from '../create-redirects'

import { createPluginData } from './helpers'

jest.mock(`fs-extra`, () => {
  const actualFsExtra = jest.requireActual(`fs-extra`)
  return {
    ...actualFsExtra,
  }
})

const redirects = [
  {
    fromPath: '/foo',
    isPermanent: false,
    ignoreCase: true,
    redirectInBrowser: false,
    toPath: '/bar',
  },
  {
    fromPath: '/dogs/*',
    isPermanent: false,
    ignoreCase: true,
    redirectInBrowser: false,
    toPath: '/animals/*',
  },
  {
    fromPath: '/baz',
    isPermanent: true,
    ignoreCase: true,
    redirectInBrowser: false,
    toPath: '/qux',
  },
  {
    fromPath: '/a',
    isPermanent: false,
    ignoreCase: true,
    redirectInBrowser: true,
    toPath: '/b',
  },
]

describe(`create-redirects`, () => {
  let reporter: any

  beforeEach(() => {
    reporter = {
      info: jest.fn(),
      panic: jest.fn(),
    }
  })

  it(`writes file`, async () => {
    const pluginData = await createPluginData()

    await createRedirects(pluginData, redirects, [], reporter)

    const output = await readFile(pluginData.publicFolder(`_redirects`), `utf8`)
    expect(output).toMatchSnapshot()
    expect(reporter.panic).not.toHaveBeenCalled()
  })
})
