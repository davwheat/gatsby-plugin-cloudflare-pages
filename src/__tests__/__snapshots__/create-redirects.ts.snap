// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-redirects writes file 1`] = `
"
## Created with gatsby-plugin-cloudflare-pages
/foo  /bar  302
/dogs/*  /animals/:splat  302
/baz  /qux  301"
`;
